### PR TITLE
Initial commit to setup the GraphQL based backend

### PR DESCRIPTION
**Pull request by jagzviruz**
I have created a full fledged backend for serving data. The following features have been included.
- Apollo Server with express with clear modularized folder structure running on `http://localhost:4000/graphql`
- Modularized Type and Resolver definitions that get auto merged to makeExecutableSchema
- Server logging and basic security ernabled with CORS, Helmet, Morgan
- Hot reloading server using nodemon, capable of executing ES6 code
- Node 8 compliant
- Apollo Graphical Client that supports playground.
- Implemented dummy endpoint `/hello` to echo "Hello World!" as a proof that we can have a mix of both REST and GraphQL support.

### CODE REVIEW

**Comment by jagzviruz** [comment]
this can be a larger expansion of #1 without the docker implementation.

**Comment by taylor-lariviere** [comment]
Could we add the docker implementation from #1 ? Docker is a good thing.

**Comment by jagzviruz** [comment]
doing it now ..

**Comment by jagzviruz** [comment]
updated PR with Docker implementation.

**Review by taylor-lariviere — APPROVED** [review]
Looks like a good start, let's just get going!

File: package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "helmet": "^3.13.0",
+    "lodash": "^4.17.10",
+    "lusca": "^1.6.1",
+    "merge-graphql-schemas": "^1.5.3",
+    "morgan": "^1.9.1",
+    "path": "^0.12.7",
+    "pm2": "^3.1.2"
+  },
+  "devDependencies": {
+    "babel-cli": "^6.26.0",
```
Comment (schalela) [inline package.json]: How do we feel about Babel 7?

**Review by schalela — COMMENTED** [review]
Looking good!

File: README.md [hunk 2]
```diff
@@ -1 +1,11 @@
-# coding-coach-back-end
\ No newline at end of file
+# coding-coach-back-end
```
Comment (taylor-lariviere) [inline README.md]: This could probably be updated with the new docker workflow

**Comment by jagzviruz** [comment]
`Readme.md` updated.

**Review by taylor-lariviere — APPROVED** [review]

**Comment by jagzviruz** [comment]
making babel 7 part of this PR .. please hold off on merging.

**Comment by jagzviruz** [comment]
Code updated to use `babel 7` now.

**Comment by schalela** [comment]
👍

**Review by KaiWedekind — APPROVED** [review]
Approved

**Pull request merged**